Feat: Updated torch.nn.Modules.set_submodules()

Gunale0926 opened the pull request:
modified:   torch/nn/modules/module.py

Implemented feature request by #127712.

Gunale0926 committed:
Feat: Updated torch.nn.Modules.set_submodules()
	modified:   torch/nn/modules/module.py

Skylion007 commented on torch/nn/modules/module.py:
```diff
@@ -665,6 +665,62 @@ def get_submodule(self, target: str) -> "Module":
+                ``nn.Module``
+        """
+        if target == "":
+            raise AttributeError("Cannot set the submodule without a target name!")
```
```suggestion
            raise ValueError("Cannot set the submodule without a target name!")
```

Gunale0926 committed:
Update torch/nn/modules/module.py

Co-authored-by: Aaron Gokaslan <aaronGokaslan@gmail.com>

Skylion007 commented:
You need to sign the CLA. Also, there aren't any tests here.

Gunale0926 committed:
Updated test and linted
	modified:   test/test_nn.py
	modified:   torch/nn/modules/module.py

Gunale0926 committed:
Fix: fixed type annotation
	modified:   torch/nn/modules/module.py

Gunale0926 committed:
Lint: lint code
	modified:   torch/nn/modules/module.py

Gunale0926 committed:
Update test
	modified:   test/test_nn.py

Gunale0926 committed:
Update doc
	modified:   torch/nn/modules/module.py

mikaylagawarecki reviewed (approved):
Thanks

Gunale0926 committed:
Fix test bug

Gunale0926 committed:
Update test_nn.py

mikaylagawarecki commented:
@pytorchbot merge

mikaylagawarecki commented on test/test_nn.py:
```diff
@@ -1434,6 +1434,20 @@ def test_add_module(self):
+        net.set_submodule(target, l2)
+        self.assertEqual(net.get_submodule(target), l2)
+        self.assertRaises(ValueError, net.set_submodule, "", l)
+        self.assertRaises(AttributeError, net.set_submodule, "a.l", l)
+        self.assertRaises(AttributeError, net.set_submodule, "t.l.l2", l2)
```
Hm it seems like tests are failing

Gunale0926 committed:
Update module.py

Here must be strict torch.nn.Module type, which allows add submodule belonging to it.

Gunale0926 commented:
Bug fixed.

Gunale0926 commented:
I think some checks failed not because of my modification.

mikaylagawarecki commented:
@pytorchbot merge -r

Gunale0926 committed:
Update Lint

mikaylagawarecki commented:
@pytorchbot merge -r
